UPGRADING.md

```diff
@@ -0,0 +1,164 @@
+
+#### Breaking Changes:
+
+- SQL statements that are valid SQL but have invalid predicates will fail during parse, instead of at runtime.
+- Fixed edge cases where users could perform cartesian joins. Now, one side of a join must be a column of unique values.
+- Fixed a handful of cases that could lead to non-deterministic query results:
+    - Disallowing conflicting column names when returned from a query.
+    - Not returning unnamed columns in query results.
+    - Applying deterministic ordering for joins against subqueries.
+
+### Network Migrations
+
+- Kwil has a network migration process that allows for exporting network data (CometBFT and Postgres) and importing it into a new network. This can be used to upgrade a network to a new version (assuming the old version does not use features that are incompatible with the new version) or to create a new network with the same data. - **[Learn more](https://docs.kwil.com/docs/node/network-migrations)**
+
+### Coordinated Consensus Rules Changes
+
+- Using the [extension system](https://docs.kwil.com/docs/extensions/overview), Kwil now has a system for nodes to coordinate changing consensus rules at a specific block height. This allows for a more flexible upgrades, such as changing the block size or adding new extensions. Previously, these changes could only be made via launching an entirely new network. - **[Learn more](https://docs.kwil.com/docs/extensions/fork-background)**
+- Note: Consensus rule changes are still experimental and may change depending on feedback.
+
+### JSON-RPC Server
+
+- Kwil now includes a JSON-RPC server that allows for querying the Kwil node using JSON-RPC.
+- The JSON-RPC listen address can be set with `jsonrpc_listen_addr` in the node's `config.toml` file. By default, the JSON-RPC server listens on port 8484.
+- Kwil CLI, GO SDK, and JS SDK have been updated to use the JSON-RPC server.
+
+#### Breaking Changes:
+
+- The HTTP server (`http_listen_addr`) is DEPRECATED and will be removed in the susbequent release.
+- The gRPC server (`grpc_listen_addr`) is no longer exposed.
+
+### Node Configuration Changes
+
+In the node's `config.toml` file, the following changes have been made:
+
+- Added `app.rpc_timeout`, which imposes a timeout on RPC requests. The default is 45 seconds.
+- Added `app.db_read_timeout`, which imposes a timout on read-only DB transactions (i.e. actions/procedures with a [`view` tag](https://docs.kwil.com/docs/kuneiform/supported-features#access-tags) or an ad-hoc SELECT query). The default 5 seconds.
+- Added `rpc.broadcast_tx_timout`, which imposes a timeout for awaiting transaction confirmation when using the `--sync` flag in `kwil-cli`. The default is 10 seconds.
+- Added `app.snapshots` to support snapshot creation. See the `StateSyncConfig` struct for more information.
+- Added `app.snapshot_file` to suport starting a network with a state at genesis. This is to support the Network Migrations feature.
```

Comment:
should we rename it to genesis_snapshot_file or something like that to remove confusion with the other snapshots?

Reply:
I think that could make sense cc @brennanjl.

Reply:
Or `genesis_state`

Reply:
Sorry did not see this. I think something involving genesis makes more sense here. I actually got it confused with the other snapshot path once

Reply:
@KwilLuke can you update the doc here?